File: Archives/Problem 1/Solutions - JavaScript/Problem 1 - Solution 7.js
function findSumMultiple(multipleNumber, limitNumber) {
    limitNumber = (limitNumber - (limitNumber % multipleNumber)) / multipleNumber;
    return multipleNumber * limitNumber * (limitNumber + 1) / 2;
}

function findSumMultiples(limitNumber) {
    return  findSumMultiple(3, limitNumber - 1) +
            findSumMultiple(5, limitNumber - 1) -
            findSumMultiple(15, limitNumber - 1);
}

console.log('The sum of the multiples of 3 and 5 below 10 is:', findSumMultiples(10));
console.log('The sum of the multiples of 3 and 5 below 1000 is:', findSumMultiples(1000));

File: Archives/Problem 1/Solutions - Python/Problem 1 - Solution 7.py
def findSumMultiple(multipleNumber, limitNumber):
    limitNumber = (limitNumber - (limitNumber % multipleNumber)) / multipleNumber
    return int(multipleNumber * limitNumber * (limitNumber + 1) / 2)


def findSumMultiples(limitNumber):
    return  findSumMultiple(3, limitNumber - 1) + \
            findSumMultiple(5, limitNumber - 1) - \
            findSumMultiple(15, limitNumber - 1)


print('The sum of the multiples of 3 and 5 below 10 is:', findSumMultiples(10))
print('The sum of the multiples of 3 and 5 below 1000 is:', findSumMultiples(1000))
